independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/observers/request/RequestInObserverNotificationTest.java

```diff
@@ -32,26 +32,26 @@ public class RequestInObserverNotificationTest {
     @Test
     public void testObserverNotification() {
         ArcContainer container = Arc.container();
-        AtomicReference<String> msg = new AtomicReference<String>();
+        ReferenceWrapper msg = new ReferenceWrapper();
         RequestFoo.DESTROYED.set(false);
 
         // Request context should be activated automatically
-        container.beanManager().getEvent().select(AtomicReference.class).fire(msg);
-        String fooId1 = msg.get();
+        container.beanManager().getEvent().select(ReferenceWrapper.class).fire(msg);
```

Comment:
Can't we just use something like `container.beanManager().getEvent().select(new TypeLiteral<AtomicReference<String>>(){}).fire(msg);` instead?